torchtune/modules/transforms/tokenizers/_hf_tokenizer.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import json
+from typing import Any, Dict, List, Optional
+
+from tokenizers import Tokenizer
+from torchtune.modules.transforms.tokenizers._utils import BaseTokenizer
+
+
+class HuggingFaceTokenizer(BaseTokenizer):
+    """
+    A wrapper around Hugging Face tokenizers. See https://github.com/huggingface/tokenizers
+    This can be used to load from a Hugging Face tokenizer.json file into a torchtune BaseTokenizer.
+
+    This class will load the tokenizer.json file from tokenizer_json_path. It will
+    attempt to infer BOS and EOS token IDs from config.json if possible, and if not
+    will fallback to inferring them from generation_config.json.
+
+    Args:
+        tokenizer_json_path (str): Path to tokenizer.json file
+        tokenizer_config_json_path (Optional[str]): Path to tokenizer_config.json file. Default: None
+        generation_config_path (Optional[str]): Path to generation_config.json file.
+            Default: None
+
+    Raises:
+        ValueError: If neither tokenizer_config_json_path or generation_config_path are specified.
+    """
+
+    def __init__(
+        self,
+        *,
+        tokenizer_json_path: str,
```

Comment:
Could've sworn I made a comment here, but b/c tokenizer_json_path is required, it can be a positional argument. My thinking is that kwargs would apply to tokenizer_config_json_path and generation_config_path

Reply:
Ah duh, good point. I should've thought about it more carefully anyways